domain/transfer.go

```diff
@@ -171,3 +180,42 @@ func (TransferProcessHandler) HandleEvent(
 
 	return nil
 }
+
+// HandleTimeout handles a timeout message that has been routed to this handler.
+func (TransferProcessHandler) HandleTimeout(
+	ctx context.Context,
+	s dogma.ProcessTimeoutScope,
+	m dogma.Message,
+) error {
+	switch x := m.(type) {
+	case TransferAtScheduledTimeout:
+		s.ExecuteCommand(commands.DebitAccount{
+			TransactionID:   x.TransactionID,
+			AccountID:       x.FromAccountID,
+			TransactionType: messages.Transfer,
+			Amount:          x.Amount,
+			ScheduledDate:   x.ScheduledDate,
+		})
+
+	default:
+		panic(dogma.UnexpectedMessage)
+	}
+
+	return nil
+}
+
+// TimeoutHint returns a duration that is suitable for computing a deadline for
+// the handling of the given message by this handler.
+func (TransferProcessHandler) TimeoutHint(m dogma.Message) time.Duration {
```

Comment:
This method can be removed once `NoTimeoutHintBehavior` is re-added.

I realise the distinction between a "timeout message" and the message handling "timeout hint" is confusing. We had a big discussion about it at work if you recall, and couldn't come up with anything better.  I am very reluctant to change name of "timeout messages" at this point but we could still change the `TimeoutHint` naming if something better comes along.

Reply:
The name change sounds like a good idea. I think you created a task to discuss that more and/or rename it.